client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -954,6 +954,41 @@ public InputStream getContent() throws IOException {
             }
             return out.asInput();
         }
+
+        /**
+         * A gzip compressing enrity doesn't worked with chunked encoding with sigv4
+         *
+         * @return false
+         */
+        @Override
+        public boolean isChunked() {
+            return false;
+        }
+
+        /**
+         * A gzip entity require to content length in http headers
+         * as it doesn't work with chunked encoding for sigv4
+         *
+         * @return content lenght of gzip entity
+         */
+        @Override
+        public long getContentLength() {
+            long size = 0;
+            int chunk = 0;
+            byte[] buffer = new byte[1024];
+
+            try {
+                InputStream is = getContent();
+
+                while ((chunk = is.read(buffer)) != -1) {
+                    size += chunk;
+                }
+            } catch (Exception ex) {
+                throw new RuntimeException("failed to get compressed content lenght: " + ex.getMessage());
```

Comment:
there's a typo in `length`

Do we need to catch and re-raise a runtime exception here? Don't we want a better exception from the underlying stream error?

Reply:
See if there's an exception that can wrap the underlying exception rather than re-throw an unrelated one without a stack?

Reply:
ack will update in next commit